dispatcher_test.go

```diff
@@ -60,6 +60,44 @@ func basicDispatcher(t testing.TB) *Dispatcher {
 	return NewDispatcher(basicConfig(t))
 }
 
+func TestDispatcherNamePanic(t *testing.T) {
+	tests := []struct {
+		name        string
+		serviceName string
+	}{
+		{
+			name: "no service name",
+		},
+		{
+			name:        "invalid service name",
+			serviceName: "--",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer func() {
+				require.NotNil(t, recover(), "expected to panic")
```

Comment:
in hindsight would have been nice to return an error here so that it could be propagated to Fx during lifecycle start

Reply:
I think there's a `require.Panic` or something of the sort